packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.tsx

```diff
@@ -312,6 +317,10 @@ export class DetailsList extends BaseComponent<IDetailsListProps, IDetailsListSt
 
     const rowCount = (isHeaderVisible ? 1 : 0) + GetGroupCount(groups) + (items ? items.length : 0);
 
+    if (enableShimmer) {
+      this._shimmerInitialItems = new Array(SHIMMER_INITIAL_ITEMS);
+    }
```

Comment:
this is done for every `render` call?

Reply:
Moved it in the constructor. Good catch... It was called on every render )